hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -591,6 +591,10 @@ private void revertCompleteToInflight(HoodieInstant completed, HoodieInstant inf
     }
   }
 
+  private Path getInstantFileNamePath(String fileName) {
+    return new Path(fileName.contains(SAVE_SCHEMA_ACTION) ? metaClient.getSchemaFolderName() : metaClient.getMetaPath(), fileName);
```

Comment:
if we treat something as an instant, splitting this apart into folders seem a bit odd to me

Reply:
Yes, currently our history schema is saved in/ hoodie/.  In the schema directory, we need to save the schema through the metatable later,  once done we don't need such strange logic anymore